Objects/dictobject.c

```diff
@@ -1482,7 +1609,8 @@ dictresize(PyInterpreterState *interp, PyDictObject *mp,
     Py_ssize_t numentries = mp->ma_used;
 
     if (oldvalues != NULL) {
-         PyDictUnicodeEntry *oldentries = DK_UNICODE_ENTRIES(oldkeys);
+        ASSERT_KEYS_LOCKED(oldkeys);
```

Comment:
There are some callers to `dictresize` that do not lock the keys. For example, here's a test case that triggers the assertion via `dict_dict_merge`.

https://gist.github.com/colesbury/ec8c94b5b24ffc4b5e7a6e5b375e8995

Reply:
Ahh indeed, it'll be better to lock here instead of in the callers, and it also gets rid of the unnecessary incref.